### PR TITLE
precondition validation are added to book and user setters

### PR DESCRIPTION
**Pull request by YesidHoyosWolox**
## Summary
- Precondition validation are added to book and user setters
## Trello Card
https://trello.com/c/PRmyVhfY/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

**Review by pachecowolox — APPROVED** [review]

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -108,6 +114,7 @@ public String getYear() {
     }
 
     public void setYear(String year) {
+        Preconditions.checkNotNull(year, "Illegal Argument, year is Null.");
```
Comment (julianvasquez-wolox) [inline src/main/java/wolox/training/models/Book.java:118]: Another good precondition is to check the year is valid, like it's not from the future or a negative year.

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -117,13 +124,15 @@ public int getPages() {
 
     public void setPages(int pages) {
         this.pages = pages;
+        Preconditions.checkNotNull(pages, "Illegal Argument, pages is Null.");
```
Comment (julianvasquez-wolox) [inline src/main/java/wolox/training/models/Book.java:130]: same here, negative pages shouldn't be allowed

**Review by julianvasquez-wolox — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/models/Book.java (hunk 1 above)
Comment (YesidHoyosWolox) [reply src/main/java/wolox/training/models/Book.java:118]: Fixed at https://github.com/wolox-training/yh-java/pull/11/commits/6d7dd31f8f886894e790e10b107b0968347e49a7 commit

File: src/main/java/wolox/training/models/Book.java (hunk 2 above)
Comment (YesidHoyosWolox) [reply src/main/java/wolox/training/models/Book.java:130]: Fixed at https://github.com/wolox-training/yh-java/pull/11/commits/6d7dd31f8f886894e790e10b107b0968347e49a7 commit

**Review by julianvasquez-wolox — APPROVED** [review]

**Pull request merged**